packages/grafana-ui/src/types/events.ts

```diff
@@ -0,0 +1,6 @@
+import { eventFactory } from '@grafana/data';
+import { DataQueryResponseData, DataQueryError } from '.';
+
+export const dataReceived = eventFactory<DataQueryResponseData[]>('data-received');
+export const dataError = eventFactory<DataQueryError>('data-error');
+export const dataSnapshotLoad = eventFactory<DataQueryResponseData[]>('data-snapshot-load');
```

Comment:
there is only single namespace for @grafana/ui, and these are only for old angular plugins and something we want to migrate away from. 

maybe we can add a legacy prefix?

Reply:
or `panelEvent` like panelEventDataReceived

Reply:
Could hide them behind an `AppEvents` namespace?

Reply:
what do you mean? (we can't use typescript namespaces, they will be deprecated I think). 

Do you mean exporting an object named AppEvents with static properties? 

like 
```
export const AppEvents = {
  dataReceived: ...
}```

Think it would be PanelEvents =  ..
 in that case.

Reply:
Also, since we want to move events away from @grafana/data and to move those that are needed to @grafana/runtime, would that mean introducing a dependency on @grafana/runtime in @grafana/ui, since the ui package has its own events and depends on `eventFactory`? Or do you think it would be better to have `eventFactory` in @grafana/data since @grafana/ui already has a dependency on that package

Reply:
ah, did not think about that. @grafana/ui should not have a dependency on @grafana/runtime. 

The DataQueryXXXX types should really be in @grafana/data we have just not moved them yet. But for now let's keep the eventFactory in @grafana/data.

Reply:
`@grafana/data` makes sense to me -- we could use somethign like the approach we have for dateMath to give things a "namespace"

https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/datetime/index.ts#L6

Reply:
Yes, I meant a namespace in the sense of an object with the the events as properties, sorry for the confusion. Would that work for you, @torkelo ?

Reply:
Think these PanelEvents needs to be defined in @grafana/ui until we move the DataQueryResponse types

Reply:
> meant a namespace in the sense of an object with the the events as properties

that will work :+1: